### DISCUSSION TITLE
How can I find the supported browser version for Next.js 12.3? I have a requirement to build an app on Firefox 52.

### DISCUSSION THREAD

**Discussion by khanhduy-14** [vercel/next.js#50948] [Help] [answered]
### Summary

I want to build a Next.js 12.3 project supported by the browser versions in the image below. I have checked https://nextjs.org/docs/architecture/supported-browsers, but I think it is for Next.js 13.

This is the requirement for supported browser versions when building the project:
![MicrosoftTeams-image](https://github.com/vercel/next.js/assets/101503338/adc7f5fc-8f64-43a2-b7e8-c4a619fd03eb)

How can I find the supported browser versions for Next.js 12.3?

**Comment by icyJoseph** [comment, marked as answer]
> After investigating a bit and building an answer, I was not very satisfied, so I had intended to not comment, but I kind of want to see if someone else can give an actual version lower bound, not just for FF, but every other browser. Maybe you can continue on with the info below.

I think this could be a starting point: https://github.com/vercel/next.js/discussions/33227 it marks the switch which dropped IE11, and started to target modern browsers by default, whilst keeping opt-in for backwards compatibility. I guess you can at least try to target ff 52 there https://nextjs.org/docs/architecture/supported-browsers#browserslist

According to caniuse, https://caniuse.com/?search=es5, FF22 already had great ES5 support. And Next.js does a lot of CSS fixing up as well. 

I cannot give you a number, though but it looks very much doable. Given that IE11 was the usual suspect. I can imagine a few things might not work, but those ought to be fixable.

Since Next.js was released in 2016, and FF 52 is a 2017 release, I think you can make a pretty good bet right there. I think they also run some tests in various browsers through browserstack.

I think, unless you have access to these browser versions, there might be some rough edges, or quirks that you are not fore-seeing at this point. 

Once I worked fixing issues that arrived through Sentry, on browsers I just couldn't access, every single one of them we fixed by either dropping "modern" APIs or just bringing in a poly/pony-fill.
